<launch>
    <arg name="MAP_FRAME" default="map" />
    <arg name="CAMERA_FRAME" default="zed2_left_camera_frame" />
    <arg name="BASE_FRAME" default="base_link" />
    <arg name="map_to_odom" default="true"/>
    <arg name="odom_to_base_footprint" default="true"/>
    <arg name="POINT_CLOUD_TOPIC" default="/zed2/zed_node/point_cloud/ds_cloud_registered" />
    <arg name="CLUSTERS_PER_OBJECT" default="3" />
    <arg name="SAVE_IMAGE" default="true" />
    <node 
        pkg="tf" 
        type="static_transform_publisher" 
        name="map_broadcaster" 
        args="0 0 0 0 0 0 1 map odom 20" 
        if="$(arg map_to_odom)"
    />
    <node 
        pkg="tf" 
        type="static_transform_publisher" 
        name="odom_broadcaster" 
        args="0 0 0 0 0 0 1 odom base_footprint 20"
        if="$(arg odom_to_base_footprint)"
    />
    <node name="getObb" pkg="object_detector_3d" type="getObb" respawn="true" output="screen"/>
    
    <node name="Detection3D" pkg="object_detector_3d" type="Detection3D" respawn="true" output="screen">
        <param name="POINT_CLOUD_TOPIC" value="$(arg POINT_CLOUD_TOPIC)"/>
        <param name="MAP_FRAME" value="$(arg MAP_FRAME)" />
        <param name="BASE_FRAME" value="$(arg BASE_FRAME)" />
        <param name="CAMERA_FRAME" value="$(arg CAMERA_FRAME)" />
    </node>
    <node name="Detection3DPlace" pkg="object_detector_3d" type="Detection3DKmeans" respawn="true" output="screen">
        <param name="POINT_CLOUD_TOPIC" value="$(arg POINT_CLOUD_TOPIC)"/>
        <param name="MAP_FRAME" value="$(arg MAP_FRAME)" />
        <param name="BASE_FRAME" value="$(arg BASE_FRAME)" />
        <param name="CAMERA_FRAME" value="$(arg CAMERA_FRAME)" />
    </node>
    <node name="Clustering_Service" pkg="object_detector_3d" type="Clustering_Server.py" respawn="true" output="screen">
        <param name="CLUSTERS_PER_OBJECT" value="$(arg CLUSTERS_PER_OBJECT)" />
        <param name="SAVE_IMAGE" value="$(arg SAVE_IMAGE)" />
    </node>
</launch>
